experimental/devices/ads1x15/ads1x15.go

```diff
@@ -0,0 +1,392 @@
+// diff can be:
+// * Channel 0 - channel 1
+// * Channel 0 - channel 3
+// * Channel 1 - channel 3
+// * Channel 2 - channel 3
+func (d *Dev) PinForDifferenceOfChannels(channelA int, channelB int, maxVoltage physic.ElectricPotential, minimumFrequency physic.Frequency) (pin AnalogPin, err error) {
+	var mux int
+
+	if err = d.checkChannel(channelA); err != nil {
+		return
+	}
+	if err = d.checkChannel(channelB); err != nil {
+		return
+	}
+
+	if channelA == Channel0 && channelB == Channel1 {
+		mux = 0
+	} else if channelA == Channel0 && channelB == Channel3 {
+		mux = 1
+	} else if channelA == Channel1 && channelB == Channel3 {
+		mux = 2
+	} else if channelA == Channel2 && channelB == Channel3 {
+		mux = 3
+	} else {
+		err = errors.New("Only some differences of channels are allowed:  0 - 1, 0 - 3, 1 - 3 or 2 - 3")
+		return
+	}
+
+	return d.prepareQuery(mux, maxVoltage, minimumFrequency)
+}
+
+func (d *Dev) prepareQuery(mux int, maxVoltage physic.ElectricPotential, minimumFrequency physic.Frequency) (pin AnalogPin, err error) {
+	// Determine the most appropriate gain
+	gain, err := d.bestGainForElectricPotential(maxVoltage)
+	if err != nil {
+		return
+	}
+
+	// Validate the gain.
+	gainConf, ok := d.gainConfig[gain]
+	if !ok {
+		err = errors.New("Gain must be one of: 2/3, 1, 2, 4, 8, 16")
+		return
+	}
+
+	// Determine the voltage multiplier for this gain
+	voltageMultiplier, ok := d.gainVoltage[gain]
+	if !ok {
+		err = errors.New("Gain must be one of: 2/3, 1, 2, 4, 8, 16")
+		return
+	}
+
+	// Determine the most appropriate data rate
+	dataRate, err := d.bestDataRateForFrequency(minimumFrequency)
+	if err != nil {
+		return
+	}
+
+	dataRateConf, ok := d.dataRates[dataRate]
+
+	if !ok {
+		// Write a nice error message in case the data rate is not found
+		keys := []int{}
+		for k := range d.dataRates {
+			keys = append(keys, k)
+		}
+
+		err = fmt.Errorf("Invalid data rate. Accepted values: %d", keys)
+		return
+	}
+
+	// Build the configuration value
+	var config uint16
+	config = ads1x15ConfigOsSingle // Go out of power-down mode for conversion.
+	// Specify mux value.
+	config |= uint16((mux & 0x07) << ads1x15ConfigMuxOffset)
+	// Validate the passed in gain and then set it in the config.
+	config |= gainConf
+	// Set the mode (continuous or single shot).
+	config |= ads1x15ConfigModeSingle
+
+	// Set the data rate (this is controlled by the subclass as it differs
+	// between ADS1015 and ADS1115).
+	config |= dataRateConf
+	config |= ads1x15ConfigCompQueDisable // Disable comparator mode.
+
+	// Build the query to the ADC
+	configBytes := make([]byte, 2)
+	binary.BigEndian.PutUint16(configBytes, config)
+	preparedQuery := append([]byte{ads1x15PointerConfig}, configBytes...)
+
+	// The wait for the ADC sample to finish is based on the sample rate plus a
+	// small offset to be sure (0.1 millisecond).
+	waitTime := time.Duration(1000000/dataRate+100) * time.Microsecond
```

Comment:
Please rewrite as:
```
waitTime := (time.Second + dataRate/2) / time.Duration(dataRate) + 100* time.Microsecond
```

So you use constants instead of numbers, especially that everything is scaled in Microsecond, which makes it harder to read. My version includes better integer rounding.